app/src/main/java/com/sopt/now/gs/feature/home/HomeFragment.kt

```diff
@@ -1,23 +1,180 @@
+            }
+        })
+        initBottomBannerJob()
+    }
+
+    private fun initBottomBannerJob() {
+        bottomBannerJob = lifecycleScope.launch {
+            while (true) {
+                delay(2000)
+                if (bottomBannerCurrentPosition == 3) {
+                    binding.vpMainBottomBanner.setCurrentItem(0, true)
+                } else {
+                    binding.vpMainBottomBanner.setCurrentItem(bottomBannerCurrentPosition++, true)
+                }
+            }
+        }
+    }
+
+    private fun initMonthEvent() {
+        val EventItemList: List<HomeBanner> = listOf(
+            HomeBanner(R.drawable.img_main_month_event1),
+            HomeBanner(R.drawable.img_main_month_event2),
+            HomeBanner(R.drawable.img_main_month_event3)
+        )
+        binding.vpMainMonthEvent1.adapter = eventAdapter
+        eventAdapter?.submitList(EventItemList)
+
+        binding.vpMainMonthEvent1.registerOnPageChangeCallback(object :
+            ViewPager2.OnPageChangeCallback() {
+            override fun onPageSelected(position: Int) {
+                super.onPageSelected(position)
+                monthEventCurrentPosition = position
+            }
+        })
+        initMonthEventJob()
+    }
+
+    private fun initMonthEventJob() {
+        monthEventJob = lifecycleScope.launch {
+            while (true) {
+                delay(1000)
+                if (monthEventCurrentPosition == 3) {
+                    binding.vpMainMonthEvent1.setCurrentItem(0, true)
+                } else {
+                    binding.vpMainMonthEvent1.setCurrentItem(monthEventCurrentPosition++, true)
+                }
+            }
+        }
+    }
+
+    private fun initRecyclerView() {
+        //TODO
+        val eventList: List<HomeBanner> = listOf(
+            HomeBanner(R.drawable.img_main_month_event_banner1),
+            HomeBanner(R.drawable.img_main_month_event_banner2),
+            HomeBanner(R.drawable.img_main_month_event_banner3),
+            HomeBanner(R.drawable.img_main_month_event_banner1),
+            HomeBanner(R.drawable.img_main_month_event_banner2),
+            HomeBanner(R.drawable.img_main_month_event_banner3)
+        )
+        val snapHelper = LinearSnapHelper()
+        snapHelper.attachToRecyclerView(binding.vpMainMonthEvent2)
+
+        binding.vpMainMonthEvent2.adapter = adapter
+        adapter?.submitList(eventList)
+
+
+        monthEventJob2 = lifecycleScope.launch {
+            while (true) {
+                delay(1000)
+                if (monthEventCurrentPosition2 == 4) {
+                    binding.vpMainMonthEvent2.smoothScrollToPosition(0)
+                    monthEventCurrentPosition2 = 0
+                } else {
+                    binding.vpMainMonthEvent2.smoothScrollToPosition(monthEventCurrentPosition2++)
+                }
+            }
+        }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
     }
 }
```

Comment:
```suggestion
    override fun onDestroyView() {
        super.onDestroyView()
        autoScrollJob?.cancel() // 예시 네이밍
        _reqresAdapter = null  // 예시 네이밍
    }
}
```

job과 어뎁터는 onDestoryView시에 null처리 해주세요! 메모리 누수의 위험이 있습니다

Reply:
p1